framework/src/outputs/FileOutput.C

```diff
@@ -74,12 +77,24 @@ FileOutput::FileOutput(const InputParameters & parameters) :
     _file_base += buffer;
   }
 
-  // Check the file directory of file_base
+  // Check the file directory of file_base and create if needed
   std::string base = "./" + _file_base;
   base = base.substr(0, base.find_last_of('/'));
   if (access(base.c_str(), W_OK) == -1)
-    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
-
+  {
+    //Directory does not exist. Loop through incremental directories and create as needed.
+    size_t pos = 0;
+    std::string inc_path;
+    do
```

Comment:
I've reimplemented it using tokenize, and will push that update next week.

Ben

On May 27, 2016, at 10:25 PM, Derek Gaston <notifications@github.com> wrote:

In framework/src/outputs/FileOutput.C
<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_7101-23discussion-5Fr64983843&d=BQMFaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=hn5akMybrkn-1oiQB8nm_y7trT_BOQm9jBgbzQWwxXA&m=2390lbTSB23XLxMXvc47oj4i3Wv0aGWdAMe8Tt670fY&s=oIuerZsg6jeytC5Zo7zPUi_tZrhkQEk-db1oLlqr9o8&e=>
:

>    std::string base = "./" + _file_base;
>    base = base.substr(0, base.find_last_of('/'));
>    if (access(base.c_str(), W_OK) == -1)
> -    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
> -
> +  {
> +    //Directory does not exist. Loop through incremental directories and create as needed.
> +    size_t pos = 0;
> +    std::string inc_path;
> +    do

Or... boost:: filesystem (path iterators)... ;-)

You really DO like to antogonize don't you? 😄

There are already other functions in MooseUtils related to paths... this
should be one too... but why can't whatever it is be based on tokenize()?
All it needs is a boolean parameter for honoring escape characters.

Seriously though... we're overdesigning this. What the current code does is
covered perfectly by the current tokenize()

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_7101_files_bd2be34efac386b494aa5f1183bfd7561098d68d-23r64983843&d=BQMFaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=hn5akMybrkn-1oiQB8nm_y7trT_BOQm9jBgbzQWwxXA&m=2390lbTSB23XLxMXvc47oj4i3Wv0aGWdAMe8Tt670fY&s=q7S6WkbfYCIiG_hAqhP2CgBmlLg2z6cSOcIdlGo_aMs&e=>,
or mute the thread
<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe_AGTYdPk8DBMDyMrEcX0AbAJKiEnIY79uks5qF8OwgaJpZM4IoKUf&d=BQMFaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=hn5akMybrkn-1oiQB8nm_y7trT_BOQm9jBgbzQWwxXA&m=2390lbTSB23XLxMXvc47oj4i3Wv0aGWdAMe8Tt670fY&s=DW4jlm2VGq_E9cox2-TJ7taOY3sGPgDAV3-gCLWlbeU&e=>
.